scripts/test-docker

```diff
@@ -13,14 +13,15 @@ while true; do
   sleep 5
 done
 
-test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10|^23.0"
-test_docker_versions["ubuntu:20.04"]="^^19.03.15$|^20.10|^23.0"
-test_docker_versions["ubuntu:22.04"]="^20.10.1[3-9]|^20.10.2[0-9]|^23.0"
-test_docker_versions["centos:centos7"]="^19.03.15$|^20.10|^23.0"
-test_docker_versions["centos:centos8"]="^20.10.[7-9]|^20.10.1[0-9]|^20.10.2[0-9]|^23.0"
-test_docker_versions["oraclelinux:7"]="^19.03.15$|^20.10|^23.0"
-test_docker_versions["oraclelinux:8"]="^20.10|^23.0"
-test_docker_versions["rockylinux:8"]="^20.10.[8-9]|^20.10.1[0-9]|^20.10.2[0-9]|^23.0"
+test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10.([2-9]\d)$|^23.0"
```

Comment:
Nit, but do we actually need regex capturing group here (and in other occurrences below - except the one for `redhat/ubi9` where it's indeed needed)?
```suggestion
test_docker_versions["ubuntu:18.04"]="^19.03.15$|^20.10.[2-9]\d$|^23.0"
```

Reply:
Also just realized `.`s are unescaped and will match any character. But it's fine (and better for readability) - so we should leave those be.

Reply:
That is correct, we don't need the capturing group. Updated the regex.